docfx/docs/custom-converters.md

```diff
@@ -95,6 +95,50 @@ So even if you define your type `MyType` with <xref:PolyType.GenerateShapeAttrib
 
 [!code-csharp[](../../samples/cs/CustomConverters.cs#ArrayWitnessOnFormatter)]
 
+#### Caching sub-converters with ConverterContext
+
+For improved performance, custom converters can cache sub-converters by accepting a @Nerdbank.MessagePack.ConverterContext parameter in their constructor.
+The @Nerdbank.MessagePack.ConverterContext provides access to the @Nerdbank.MessagePack.ConverterContext.GetConverter* methods, allowing converters to obtain and cache sub-converters at construction time rather than looking them up on each invocation.
+
+When a converter type is registered via @Nerdbank.MessagePack.MessagePackSerializer.ConverterTypes?displayProperty=nameWithType, the serializer will automatically invoke a public constructor that accepts a @Nerdbank.MessagePack.ConverterContext parameter if one exists.
+If no such constructor is found, the serializer falls back to invoking a parameterless constructor.
+
+Here's an example:
+
+```cs
+[GenerateShapeFor<string>]
+public partial class MyConverter : MessagePackConverter<MyType>
+{
+    private readonly MessagePackConverter<string> stringConverter;
+
+    public MyConverter(ConverterContext context)
+    {
+        this.stringConverter = context.GetConverter<string>(GeneratedTypeShapeProvider);
+    }
+
+    public override void Write(ref MessagePackWriter writer, in MyType? value, SerializationContext context)
+    {
+        this.stringConverter.Write(ref writer, value?.Name, context);
+    }
+
+    public override MyType? Read(ref MessagePackReader reader, SerializationContext context)
+    {
+        string? name = this.stringConverter.Read(ref reader, context);
+        return new MyType { Name = name };
+    }
+}
```

Comment:
Please move this sample into the `samples` directory next to other C# samples and reference it from here so that docfx inlines it.